test/e2e/proxy_test.go

```diff
@@ -736,11 +736,11 @@ func setStoneSoupConfig(t *testing.T, hostAwait *wait.HostAwaitility, memberAwai
 func verifyHasExpectedWorkspace(t *testing.T, expectedWorkspace toolchainv1alpha1.Workspace, actualWorkspaces ...toolchainv1alpha1.Workspace) {
 	for _, actualWorkspace := range actualWorkspaces {
 		if actualWorkspace.Name == expectedWorkspace.Name {
-			assert.Equal(t, expectedWorkspace.Status, actualWorkspace.Status)
-			assert.Equal(t, expectedWorkspace.ObjectMeta.ResourceVersion, actualWorkspace.ObjectMeta.ResourceVersion)
-			assert.Equal(t, expectedWorkspace.ObjectMeta.Generation, actualWorkspace.ObjectMeta.Generation)
-			assert.Equal(t, expectedWorkspace.ObjectMeta.CreationTimestamp, actualWorkspace.ObjectMeta.CreationTimestamp)
-			assert.Equal(t, expectedWorkspace.ObjectMeta.UID, actualWorkspace.ObjectMeta.UID)
+			assert.NotEmpty(t, actualWorkspace.Status, "Workspace.Status field is empty", actualWorkspace, expectedWorkspace)
```

Comment:
We can't just check if the status is not empty. We need to check the actual values. The status has the list of namespaces and other critical information we need to test.

Reply:
makes sense, I've restored the check in [ad0fb7e](https://github.com/codeready-toolchain/toolchain-e2e/pull/668/commits/ad0fb7e58930af89a4713957f287010f1a66888f)

Reply:
As I was trying to understand in the discussion on the slack channel, since the tests is first fetching the list of workspaces https://github.com/codeready-toolchain/toolchain-e2e/blob/614010ed3f9ce916589dc6d5036fa143b9f381ac/test/e2e/proxy_test.go#L384 
and then waiting for the space to match condition https://github.com/codeready-toolchain/toolchain-e2e/blob/614010ed3f9ce916589dc6d5036fa143b9f381ac/test/e2e/proxy_test.go#L751

I am afraid, that by the time the Space is retrieved, it may have a difference also in the `.Status` object. But maybe that's less frequent or I misunderstood how it works. 🤷‍♂️

Reply:
The Workspace.Status mirrors the following values from the Space resource: `Namespaces`, `Owner`, `Role`, and Type. The test calls `VerifyResourcesProvisionedForSignup` after creating each user and does not modify any of those Space values so I believe none of those should change after `listWorkspaces` is called. So it should be safe to keep the check.

The conversation on slack was more specifically about the `ObjectMeta` fields because those can be changed in many cases and is not really possible for us to predict so it's better to just check that they're not empty.

Reply:
Ok, thanks for the explanation, and makes sense that not changing the users after creation may not modify the Status object. I wanted to perform some other test runs, before I merge, so I'll push few "empty" commits into this PR.